src/components/new-safe/create/logic/index.ts

```diff
@@ -238,8 +241,11 @@ export const createNewUndeployedSafeWithoutSalt = (
       threshold: safeAccountConfig.threshold,
       owners: safeAccountConfig.owners,
       fallbackHandler: fallbackHandlerAddress,
-      to: includeMigration ? SAFE_TO_L2_SETUP_ADDRESS : ZERO_ADDRESS,
-      data: includeMigration ? SAFE_TO_L2_SETUP_INTERFACE.encodeFunctionData('setupToL2', [safeL2Address]) : EMPTY_DATA,
+      to: includeMigration && safeToL2SetupAddress ? safeToL2SetupAddress : ZERO_ADDRESS,
+      data:
+        includeMigration && safeToL2SetupInterface
```

Comment:
Nit: the `safeToL2SetupInterface` cannot be undefined.

```suggestion
        includeMigration
```